Cargo.lock

```diff
@@ -93,6 +95,12 @@ version = "1.0.1"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "cdb031dd78e28731d87d56cc8ffef4a8f36ca26c38fe2de700543e627f8a464a"
 
+[[package]]
```

Comment:
In reviewing this, it looks to me like the Cargo.lock delta is larger than it needs to be - it's the result of a full `cargo update` rather than just a `cargo check`? That being said, when I tried to regenerate this `Cargo.Lock` file to get the more minimal set of changes I got the following:
```
error: failed to select a version for `ff`.
    ... required by package `jubjub v0.7.0 (https://github.com/zkcrypto/jubjub?rev=ab1f7647acc32ab597f2ec1f5b80a45afca90e59#ab1f7647)`
    ... which is depended on by `librustzcash v0.2.0 (/home/nuttycom/work/zcash/canon)`
versions that meet the requirements `^0.10.1` are: 0.10.1

all possible versions conflict with previously selected packages.

  previously selected package `ff v0.10.0`
    ... which is depended on by `bellman v0.10.0`
    ... which is depended on by `librustzcash v0.2.0 (/home/nuttycom/work/zcash/canon)`

failed to select a version for `ff` which could resolve this conflict
```

That suggests to me that there is some inconsistency in the `patch.crates-io` section of the Cargo.toml file; I'm pretty sure it's possible to choose a set of versions where these dependencies are consistent?

Reply:
I regenerated the more minimal set locally, and confirmed this diff is larger. But AFAICT from a glance the differences should be harmless.